### PR TITLE
Add "Kaomoji Search" extension

### PR DESCRIPTION
**Pull request by yalishanda42**
## Description

This PR aims to add a new extension called Kaomoji Search. It allows to people to search for an ascii/unicode emote and copy it to the clipboard.

## Type of change

- New extension

## Screencast

demo

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

# Note

I currently added the extension as a submodule as I had it initially hosted as a repo in my GitHub profile. If that is not desired, I have no problem just copying the project into the `extensions` directory instead.

### CODE REVIEW

**Comment by allexks** [comment]
Alright, sorry for the failing pipeline, I will just copy it into the project. Just to ask though, is it on purpose that submodules are not supported?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @allexks 👋 

Thanks for your contribution 💪 

Some feedback for you to address:
- If searching for, e.g., “Bear”, no matches found, only “bear” works.
- Move folder with preview to assets

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @allexks 👋 

Much better 💪!

Now it's possible to search a bit better, however, when using space it all breaks. Shouldn't it be possible to search for, e.g., “Flower girl”?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by allexks** [comment]
Thanks for the feedback! I reimplemented the searching algorithm now, results should be better.

**Review by pernielsentikaer — APPROVED** [review]
Hi @allexks 👋 

Thanks, MUCH better! Let's proceed to the last stage 🚀 I'm going to accept it now, my colleague will do a final review before merging 💎

**Comment by pernielsentikaer** [comment]
Hi @allexks 👋 

Just found a minor thing. Can you provide a light/dark icon? Otherwise, this won't look good in different themes. You can find more info [here](https://developers.raycast.com/information/manifest).

**Comment by allexks** [comment]
Thanks, I added a brighter version for the dark theme. I don't quite like the way this icon looks like when small, to be honest, but maybe in the future it can be improved, I just don't have the skills to come up with and create something better for now.

File: extensions/kaomoji-search/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  }
+
+  return {
+    state: state,
+    search: search,
+  };
+}
+
+async function performSearch(searchText: string, signal: AbortSignal): Promise<SearchResult[]> {
+  console.log("searching for", searchText);
```
Comment (sxn) [inline extensions/kaomoji-search/src/index.tsx:77]:
```suggestion
```

File: extensions/kaomoji-search/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  const results = await searchForResults(searchText);
+
+  if (signal.aborted) {
+    return Promise.reject(new AbortError());
+  }
+
+  console.log("results count", results.length);
```
Comment (sxn) [inline extensions/kaomoji-search/src/index.tsx:85]:
```suggestion
```

**Comment by sxn** [comment]
thanks for the work @allexks! 🎉 

Playing around with the extension, I kept wanting to be able to paste a kaomoji instead of copying to clipboard. Do you think it makes sense to add a [PasteAction](https://developers.raycast.com/api-reference/user-interface/actions#pasteaction) in addition to [CopyToClipboardAction](https://developers.raycast.com/api-reference/user-interface/actions#copytoclipboardaction)? 🤔

**Comment by pernielsentikaer** [comment]
I could have sworn that I asked the same question earlier, but I guess I only dreamt it! I think Paste would be awesome 💪

**Comment by allexks** [comment]
@sxn @pernielsentikaer Oh, absolutely, I thought the same while creating it but for some reason didn't think that it was a builtin action 😄  I made it the default now, thanks for the suggestion.

**Pull request merged**